juntagrico/lifecycle/sub.py

```diff
@@ -22,19 +22,40 @@ def sub_pre_save(sender, instance, **kwargs):
 
 def handle_sub_activated(sender, instance, **kwargs):
     instance.activation_date = instance.activation_date or timezone.now().date()
+    change_date = instance.activation_date
+    for part in instance.future_parts.all():
+        part.activate(change_date)
     for member in instance.recipients:
         if member.subscription_current is not None:
             raise ValidationError(
                 _('Ein Bezüger hat noch ein/e/n aktive/n/s {0}').format(Config.vocabulary('subscription')),
                 code='invalid')
+    for sub_membership in instance.subscriptionmembership_set.all():
+        sub_membership.join_date = change_date
+        sub_membership.save()
 
 
 def handle_sub_deactivated(sender, instance, **kwargs):
     instance.deactivation_date = instance.deactivation_date or timezone.now().date()
+    change_date = instance.deactivation_date
+    for extra in instance.extra_subscription_set.all():
+        extra.deactivate(change_date)
```

Comment:
actually no
since tey are decoupled in a sense of resouces
for example if you have enough vegetables for the subscription but not enough apples for the extrasubscription one activates the main subscription but the extra subscription will not be activated until there are enough apples